src/openrct2/actions/ScenerySetRestrictedAction.cpp

```diff
@@ -0,0 +1,57 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2024 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "ScenerySetRestrictedAction.h"
+
+#include "../object/SceneryGroupObject.h"
+#include "../world/Scenery.h"
+
+ScenerySetRestrictedAction::ScenerySetRestrictedAction(ObjectType objectType, ObjectEntryIndex entryIndex, bool restricted)
+    : _objectType(objectType)
+    , _objectIndex(entryIndex)
+    , _restricted(restricted)
+{
+}
+
+void ScenerySetRestrictedAction::Serialise(DataSerialiser& stream)
+{
+    GameAction::Serialise(stream);
+
+    stream << DS_TAG(_objectType) << DS_TAG(_objectIndex) << DS_TAG(_restricted);
+}
+
+void ScenerySetRestrictedAction::AcceptParameters(GameActionParameterVisitor& visitor)
+{
+    visitor.Visit("objectType", _objectType);
+    visitor.Visit("objectIndex", _objectIndex);
```

Comment:
for all other game actions i think this is called `object`

Reply:
@Basssiiie , @Sadret is that the name you would expect to use for the action in plugins?

Reply:
I'd say consistency across the API would be preferred yes, so my preference would go to `"object"` for the Javascript side as well.

Reply:
> @Basssiiie , @Sadret is that the name you would expect to use for the action in plugins?

yes, thanks for asking